File: ios/Source/Models/SetSessionConfigMessage.swift
// Copyright (c) 2017-2019 Coinbase Inc. See LICENSE

import Foundation

struct SetSessionConfigMessage: Codable, JSONDeserializable, JSONSerializable {
    /// Type of message
    let type: ClientMessageType = .setSessionConfig

    /// Client generated request ID
    let requestId: Int32

    /// Server generated session ID
    let sessionId: String

    /// Client generated request ID
    let webhookId: String

    /// Server generated session ID
    let webhookUrl: String

    let metadata: [String: String]

    enum CodingKeys: String, CodingKey {
        case type
        case requestId = "id"
        case sessionId
        case webhookId
        case webhookUrl
        case metadata
    }
}

File: android/walletlink/src/main/java/com/coinbase/walletlink/models/SetSessionConfigMessage.kt
package com.coinbase.walletlink.models

import com.coinbase.walletlink.interfaces.JsonSerializable
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import com.squareup.moshi.Moshi

@JsonClass(generateAdapter = true)
data class SetSessionConfigMessage(
    val type: ClientMessageType = ClientMessageType.SET_SESSION_CONFIG,
    @field:Json(name = "id") val requestId: Int,
    val sessionId: String,
    val webhookId: String,
    val webhookUrl: String,
    val metadata: Map<String, String>
) : JsonSerializable {
    override fun asJsonString(): String {
        val moshi = Moshi.Builder().build() // FIXME: hish - shared?
        val adapter = moshi.adapter<SetSessionConfigMessage>(SetSessionConfigMessage::class.java)
        return adapter.toJson(this)
    }
}
